community/consistency-check/src/main/java/org/neo4j/consistency/checking/cache/CacheAccess.java

```diff
@@ -0,0 +1,258 @@
+/*
+ * Copyright (c) 2002-2015 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.consistency.checking.cache;
+
+import java.util.Collection;
+
+import org.neo4j.consistency.checking.RecordCheck;
+import org.neo4j.consistency.checking.full.CheckStage;
+import org.neo4j.consistency.report.ConsistencyReport;
+import org.neo4j.consistency.report.ConsistencyReport.PropertyConsistencyReport;
+import org.neo4j.consistency.report.ConsistencyReporter;
+import org.neo4j.consistency.statistics.Counts;
+import org.neo4j.consistency.statistics.Counts.Type;
+import org.neo4j.consistency.store.RecordAccess;
+import org.neo4j.kernel.impl.store.record.AbstractBaseRecord;
+import org.neo4j.kernel.impl.store.record.NodeRecord;
+import org.neo4j.kernel.impl.store.record.PropertyRecord;
+
+/**
+ * Just as {@link RecordAccess} is the main access point for {@link AbstractBaseRecord} and friends,
+ * so is {@link CacheAccess} the main access point for cached values related to records, most often caching
+ * parts of records, specific to certain {@link CheckStage stages} of the consistency check.
+ *
+ * The access patterns to {@link CacheAccess} is designed to have multiple threads concurrently, and so
+ * {@link #client()} provides a {@link Client} that accesses the cache on the current thread's behalf.
+ *
+ * The cache is a compact representation of records, tied to an id, for example nodeId. There can be multiple
+ * cached values per id, selected by {@code slot}.
+ */
+public interface CacheAccess
+{
+    /**
+     * Client per thread for accessing cache and counts for statistics
+     */
+    interface Client
```

Comment:
Internal classes/interfaces has its use, just like packages have. Look at what we usually do with `Adapter`, `Delegator`, `Monitor` or `Configuration`... all internal interfaces that makes those small interfaces easier to understand when they're internal, also reads better when spelling out, f.ex. `CacheAccess.Client`.